server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -44,14 +63,262 @@ public AbstractProfileBreakdown<QueryTimingType> context(Object context) {
+                );
+                queryBreakdownMap.compute(
+                    avgBreakdownTypeTime,
+                    (key, value) -> (value == null) ? sliceBreakdownTypeTime : sliceBreakdownTypeTime + value
+                );
+
+                // compute max/min/avg TimingType count across slices
+                queryBreakdownMap.compute(
+                    maxBreakdownTypeCount,
+                    (key, value) -> (value == null) ? sliceBreakdownTypeCount : Math.max(sliceBreakdownTypeCount, value)
+                );
+                queryBreakdownMap.compute(
+                    minBreakdownTypeCount,
+                    (key, value) -> (value == null) ? sliceBreakdownTypeCount : Math.min(sliceBreakdownTypeCount, value)
+                );
+                queryBreakdownMap.compute(
+                    avgBreakdownTypeCount,
+                    (key, value) -> (value == null) ? sliceBreakdownTypeCount : sliceBreakdownTypeCount + value
+                );
+
+                // query start/end time for a TimingType is min/max of start/end time across slices for that TimingType
+                queryTimingTypeEndTime = Math.max(queryTimingTypeEndTime, sliceBreakdown.getValue().get(sliceEndTimeForTimingType));
+                queryTimingTypeStartTime = Math.min(queryTimingTypeStartTime, sliceBreakdown.getValue().get(sliceStartTimeForTimingType));
+                queryTimingTypeCount += sliceBreakdownTypeCount;
             }
+            queryBreakdownMap.put(timingTypeKey, queryTimingTypeEndTime - queryTimingTypeStartTime);
+            queryBreakdownMap.put(timingTypeCountKey, queryTimingTypeCount);
+            queryBreakdownMap.compute(avgBreakdownTypeTime, (key, value) -> (value == null) ? 0 : value / sliceLevelBreakdowns.size());
+            queryBreakdownMap.compute(avgBreakdownTypeCount, (key, value) -> (value == null) ? 0 : value / sliceLevelBreakdowns.size());
+            // compute query end time using max of query end time across all timing types
+            queryEndTime = Math.max(queryEndTime, queryTimingTypeEndTime);
         }
+        queryNodeTime = queryEndTime - createWeightStartTime;
+        return queryBreakdownMap;
+    }
+
+    @Override
+    public long toNodeTime() {
+        return queryNodeTime;
+    }
+
+    @Override
+    public void associateCollectorToLeaves(Collector collector, LeafReaderContext leaf) {
+        sliceCollectorToLeaves.computeIfAbsent(collector, k -> new ArrayList<>()).add(leaf);
+    }
+
+    @Override
+    public void associateCollectorToLeaves(Map<Collector, List<LeafReaderContext>> collectorToLeaves) {
+        this.sliceCollectorToLeaves.putAll(collectorToLeaves);
+    }
+
+    Map<Collector, List<LeafReaderContext>> getSliceCollectorToLeaves() {
+        return Collections.unmodifiableMap(sliceCollectorToLeaves);
+    }
+
+    Long getMaxSliceNodeTime() {
```

Comment:
Please use primitive types since these fields are primitives:

```suggestion
    long getMaxSliceNodeTime() {
```